### PR TITLE
docs: deprecate esm5 and fesm5 code distribution in our npm packages

### PR DESCRIPTION
**Pull request by IgorMinar**
See diff for more info.

### CODE REVIEW

File: aio/content/guide/deprecations.md [hunk 1]
```diff
@@ -393,6 +394,56 @@ export class MyModule {
 }
 ```
 
+{@a esm5-fesm5}
+### `esm5` and `fesm5` code formats in @angular/* npm packages
+
+As of Angular v8, the CLI primarily consumes the `fesm2015` variant of the code distributed via @angular/* npm packages.
+This renders the `esm5` and `fesm5` distributions oboslete and unnecessary, adding bloat to the package size and slowing down npm installations.
```
Comment (gkalpak) [inline aio/content/guide/deprecations.md]:
```suggestion
This renders the `esm5` and `fesm5` distributions obsolete and unnecessary, adding bloat to the package size and slowing down npm installations.
```

Comment (IgorMinar) [reply aio/content/guide/deprecations.md]: thanks

File: aio/content/guide/deprecations.md [hunk 2]
```diff
@@ -56,11 +56,12 @@ v9 - v12
...
 | `@angular/core`               | [`defineInjectable`](#core)                                                   | <!--v8--> v11 |
-| `@angular/core`               | [`entryComponents`](api/core/NgModule#entryComponents)                        | <!--v9--> v11 | 
-| `@angular/core`               | [`ANALYZE_FOR_ENTRY_COMPONENTS`](api/core/ANALYZE_FOR_ENTRY_COMPONENTS)       | <!--v9--> v11 | 
+| `@angular/core`               | [`entryComponents`](api/core/NgModule#entryComponents)                        | <!--v9--> v11 |
+| `@angular/core`               | [`ANALYZE_FOR_ENTRY_COMPONENTS`](api/core/ANALYZE_FOR_ENTRY_COMPONENTS)       | <!--v9--> v11 |
 | `@angular/router`             | [`loadChildren` string syntax](#loadChildren)                                 | <!--v9--> v11 |
 | `@angular/core/testing`       | [`TestBed.get`](#testing)                                                     | <!--v9--> v12 |
 | `@angular/router`             | [`ActivatedRoute` params and `queryParams` properties](#activatedroute-props) | unspecified |
+| npm package format            | [`esm5` and `fesm5` entry-points in @angular/* npm packages](guide/deprecations#esm5-fesm5) | <!-- v9 --> v10 |
```
Comment (kara) [inline aio/content/guide/deprecations.md]: This table is organized by release, so I think this line should be at the end of the v10 section (below "polyfills")

File: aio/content/guide/deprecations.md [hunk 3]
```diff
@@ -393,6 +394,56 @@ export class MyModule {
 }
 ```
 
+{@a esm5-fesm5}
+### `esm5` and `fesm5` code formats in @angular/* npm packages
+
+As of Angular v8, the CLI primarily consumes the `fesm2015` variant of the code distributed via @angular/* npm packages.
+This renders the `esm5` and `fesm5` distributions obsolete and unnecessary, adding bloat to the package size and slowing down npm installations.
+
+The future removal of this distribution, will have no impact on CLI users, unless they modified their build configuration to explicitly consume these code distributions.
```
Comment (kara) [inline aio/content/guide/deprecations.md]:
```suggestion
The future removal of this distribution will have no impact on CLI users, unless they modified their build configuration to explicitly consume these code distributions.
```

File: aio/content/guide/deprecations.md [hunk 4]
```diff
@@ -393,6 +394,56 @@ export class MyModule {
...
+
+Any application still relying on the `esm5` and `fesm5` as the input to their build system, will need to ensure that their build pipeline is capable of accepting JavaScript code confirming to EcmaScript 2015 (ES2015) language specification.
```
Comment (kara) [inline aio/content/guide/deprecations.md]:
```suggestion
Any application still relying on the `esm5` and `fesm5` as the input to their build system will need to ensure that their build pipeline is capable of accepting JavaScript code conforming to EcmaScript 2015 (ES2015) language specification.
```

File: aio/content/guide/deprecations.md [hunk 5]
```diff
@@ -393,6 +394,56 @@ export class MyModule {
...
+  "main": "./bundles/core.umd.js",
+  "module": "./fesm2015/core.js",
+  "es2015": "./fesm2015/core.js",
+  "esm2015": "./esm2015/core.js",
+  "fesm2015": "./fesm2015/core.js",
+  ...
+}
+```
+
+For more information about the npm package format please see the [Angular Package Format spec](https://goo.gl/jB3GVv).
```
Comment (kara) [inline aio/content/guide/deprecations.md]:
```suggestion
For more information about the npm package format, please see the [Angular Package Format spec](https://goo.gl/jB3GVv).
```

**Review by kara — CHANGES_REQUESTED** [review]
Content looks good, but some grammar nits

**Review by kapunahelewong — APPROVED** [review]
Changes made and pushed up. LGTM.

**Review by kara — APPROVED** [review]
LGTM

**Comment by kara** [comment]
merge-assistance: global approval

**Pull request closed without merging**